### ISSUE TITLE
Documentation for Mattersupport

### ISSUE THREAD

**Issue by muralivm** [below/MatterSample#6]
Hi, Thanks for sharing the matter sample app.

Can you please provide more info on the sample app?
1.  What is the ecosystem and could not find more info on apple. can you please help us here.
2.  Using the sample app, we are able to commission matter accessory over thread, after that we could not see the accessory in apple home or in sample app. So could you please help us to add an accessory into home app from your sample app.

**Comment by below** [comment]
My first question is: What are you trying to achieve? Also: This sample code _**does not**_ commission the device, you will have to do that yourself

**Comment by amareshkv-allegion** [comment]
Hey Alexandar,

Me and Murali were trying this out with a matter device to pair and then setup using Apple homekit and add this matter accessory to Apple Home.
So that using Apple Homekit APIs, the Matter device can be controlled using our custom app (not relying on Apple Home app, because we can provide all the features of Matter device in our app)

**Comment by below** [comment]
Hey @amareshkv-allegion @muralivm,
glad I asked the question: This code is not for you, you want to do something else.
What you want is to pair the accessory using the normal [HomeKit Setup](https://developer.apple.com/documentation/homekit/hmaccessorysetupmanager/3920432-performaccessorysetup). 

Then, you want to access the HomeKit [Shared Matter Controller](https://developer.apple.com/documentation/matter/mtrdevicecontroller/4122909-sharedcontroller) using the [matterControllerXPCConnectBlock](https://developer.apple.com/documentation/homekit/hmhome/4087760-mattercontrollerxpcconnectblock/)
After this, you should be able to access the accessories using the Matter Framework.

This is not what I am doing at all. It would be nice if you could share your findings, once you get this working

**Comment by muralivm** [comment]
Thanks for your response. 

Yeah we are trying to pair EFR32XG21 via thread framework using Mattersample application we could debug all the RequestHandler extension API's and able to see onboarding Matter payload and finally  "func configureDevice(named name: String, in room: MatterAddDeviceRequest.Room?) async"  API is invoked. After that we are not able to proceed further. 

We could not find more documentation on matter support framework.
We went though this link https://developer.apple.com/apple-home/matter/ and we started with MatterSupport framework to pair device EFR32XG21.

Since we are new to matter protocol could you please help us to understand what is your matterSample application intention and what are you demonstrating with this sample application.

Can you please pointout more documentation for Matter Pairing using MatterSupport framework or Matter Framework. 

we went though https://github.com/bzbarsky-apple/connectedhomeip/tree/master/src/darwin sample as well which supports only WIFI network pairing.

**Comment by below** [comment]
> Since we are new to matter protocol could you please help us to understand what is your matterSample application intention and what are you demonstrating with this sample application.

We have our own Matter C2-Controller, our own Thread Border Router, and our own Matter Fabric. We are using the MatterSupport Framework to Commission devices into our own Fabric.

As it sounds, you do not have any of this, and your want to control your Matter Device through HomeKit. This is a valid UseCase, but the MatterSupport Framework does not help you here.

Let me repeat this: If you are not implementing your own Matter Controller, then you do not need the MatterSupport Framework

**Comment by amareshkv-allegion** [comment]
Understood. Thanks for the info.

We are trying to understand all different scenarios which might occur for our smart devices. IN that 

Scenario 1:
I only have an Amazon Matter controller (Alexa or echo dot).
Now can i create custom app like yours, which can pair the smart device through Alexa Matter controller ? or is it possible only via Alexa iOS/Android app ?

Scenario 2:
Using your sample app, where you are pairing with your own fabric and controller, can you share the device to other controllers like Alexa, Apple Home or Smart things without using their respective apps ?

**Comment by below** [comment]
> Scenario 1: I only have an Amazon Matter controller (Alexa or echo dot). Now can i create custom app like yours, which can pair the smart device through Alexa Matter controller ? or is it possible only via Alexa iOS/Android app ?

The short: Only the Alexa App can do that.
The long: You probably are still confused about the Ecosystem. Your Echo can be a Matter Controller, or a Thread Border Router (TBR), or both. Apple HomeKit and other Ecosystems can utilize the Echo TBR _**if and only if**_ someone with access to the Active Data Set of the device has added the credentials to your iOS device. This "someone" would likely be the Alexa App.
Nobody but the Alexa App can utilize the Echo Matter Controller. Not HomeKit. Not you.
I am unsure if Amazon is planning to publish any sort of Matter API.

> Scenario 2: Using your sample app, where you are pairing with your own fabric and controller, can you share the device to other controllers like Alexa, Apple Home or Smart things without using their respective apps ?

As of today: No.
There may be additions to the Matter Standard that allow this, **but** this will not be done through the `MatterSupport.Framework`, rather through the Controller itself.

Does this help a bit?

P.S.: If you are writing the Alexa App or similar, this sample code is relevant to you.

**Comment by amareshkv-allegion** [comment]
yes understood.

**Comment by HururuekChapChap** [comment]
@below 
I found this article while searching for multi-administration of Homekit's Matter product. 
And I found out that matterControllerXPCConnectBlock API exists in Homekit Framework and tried to use it, but it failed.
How do I get the MTRDeviceController object into Homekit's Matter product?
An error message appears stating that the MTRDeviceController object must be started up.

**Comment by below** [comment]
Yes, this is an error that occurs often. I am currently investigating the correct way, and will publish my results. Stay tuned!

**Comment by HururuekChapChap** [comment]
@below
My heart is pounding.
Thank you.
